File: server/env_test.js
console.log('Test des variables d\'environnement en Node.js:');

// Test des variables d'environnement
console.log('\n1. Variables d\'environnement:');
console.log('- APP_ENV:', process.env.APP_ENV);
console.log('- CUSTOM_VAR:', process.env.CUSTOM_VAR);
console.log('- SERVER_ID:', process.env.SERVER_ID);
console.log('- SANDBOX_ROOT:', process.env.SANDBOX_ROOT);
console.log('- SCRIPT_TYPE:', process.env.SCRIPT_TYPE);

// Test des chemins
console.log('\n2. Chemins:');
console.log('- process.cwd():', process.cwd());
console.log('- __dirname:', __dirname);
console.log('- __filename:', __filename);

// Test des opérations sur les fichiers
console.log('\n3. Test des opérations fichiers:');
const fs = require('fs');
const path = require('path');

try {
    // Créer un fichier
    fs.writeFileSync('/test.txt', 'Test de fichier');
    console.log('✓ Création de fichier réussie');

    // Lire le fichier
    const content = fs.readFileSync('/test.txt', 'utf8');
    console.log('✓ Lecture de fichier réussie:', content);

    // Tester l'accès à un fichier hors sandbox
    try {
        fs.readFileSync('/../hors_sandbox.txt');
        console.log('❌ Accès non autorisé réussi');
    } catch (error) {
        console.log('✓ Accès hors sandbox bloqué:', error.message);
    }

    // Supprimer le fichier de test
    fs.unlinkSync('/test.txt');
    console.log('✓ Suppression de fichier réussie');
} catch (error) {
    console.error('❌ Erreur lors des tests fichiers:', error.message);
}

// Test des modules
console.log('\n4. Test des modules:');
try {
    require('fs');
    console.log('✓ Module fs chargé');
    require('path');
    console.log('✓ Module path chargé');
} catch (error) {
    console.error('❌ Erreur lors du chargement des modules:', error.message);
} 

File: test/env_test.py
import os
import sys

print('Test des variables d\'environnement en Python:')

# Test des variables d'environnement
print('\n1. Variables d\'environnement:')
print('- APP_ENV:', os.environ.get('APP_ENV'))
print('- CUSTOM_VAR:', os.environ.get('CUSTOM_VAR'))
print('- SERVER_ID:', os.environ.get('SERVER_ID'))
print('- SANDBOX_ROOT:', os.environ.get('SANDBOX_ROOT'))
print('- SCRIPT_TYPE:', os.environ.get('SCRIPT_TYPE'))

# Test des chemins
print('\n2. Chemins:')
print('- Dossier actuel:', os.getcwd())
print('- Dossier absolu:', os.path.abspath('.'))
print('- PYTHONPATH:', os.environ.get('PYTHONPATH'))

# Test des opérations sur les fichiers
print('\n3. Test des opérations fichiers:')
try:
    # Créer un fichier
    with open('/test.txt', 'w') as f:
        f.write('Test de fichier Python')
    print('✓ Création de fichier réussie')

    # Lire le fichier
    with open('/test.txt', 'r') as f:
        content = f.read()
    print('✓ Lecture de fichier réussie:', content)

    # Tester l'accès à un fichier hors sandbox
    try:
        with open('/../hors_sandbox.txt', 'r') as f:
            print('❌ Accès non autorisé réussi')
    except Exception as e:
        print('✓ Accès hors sandbox bloqué:', str(e))

    # Supprimer le fichier de test
    os.remove('/test.txt')
    print('✓ Suppression de fichier réussie')
except Exception as e:
    print('❌ Erreur lors des tests fichiers:', str(e))

# Test des modules
print('\n4. Test des modules:')
try:
    import json
    print('✓ Module json chargé')
    import urllib
    print('✓ Module urllib chargé')
except Exception as e:
    print('❌ Erreur lors du chargement des modules:', str(e)) 